dockers/base-conda/Dockerfile

```diff
@@ -104,16 +104,6 @@ RUN \
     pip install -r requirements-examples.txt --no-cache-dir --find-links https://download.pytorch.org/whl/test/torch_test.html && \
     rm assistant.py
 
-RUN \
-   apt-get purge -y cmake && \
-   wget -q https://github.com/Kitware/CMake/releases/download/v3.20.2/cmake-3.20.2.tar.gz && \
-   tar -zxvf cmake-3.20.2.tar.gz && \
-   cd cmake-3.20.2 && \
-   ./bootstrap -- -DCMAKE_USE_OPENSSL=OFF && \
-   make && \
-   make install && \
-   cmake  --version
```

Comment:
This cmake installation block was added by #12304, but I'm not sure why we needed it in the first place even though cmake gets installed with apt-get above.

cc @Borda if you remember any reason I'm missing :)

Reply:
I think that the latest Horovod for CUDA docker image requires CMake above .13, which is not available as compiled lib for Linus in the mainstream distributions so it needed to be compiled from source...